android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/ads/AdsTelemetry.kt

```diff
@@ -44,13 +44,14 @@ class AdsTelemetry : BaseSearchTelemetry() {
         cachedCookies = message.getJSONArray(ADS_MESSAGE_COOKIES_KEY).toList()
 
         val urls = message.getJSONArray(ADS_MESSAGE_DOCUMENT_URLS_KEY).toList<String>()
+        val uri = Uri.parse(message.getString(ADS_MESSAGE_SESSION_URL_KEY))
         val provider = getProviderForUrl(message.getString(ADS_MESSAGE_SESSION_URL_KEY))
 
         provider?.let {
             if (it.containsAdLinks(urls)) {
                 emitFact(
                     SERP_SHOWN_WITH_ADDS,
-                    it.name,
+                    getTrackKey(it, uri, cachedCookies),
```

Comment:
If the affected metric is [this one](https://dictionary.telemetry.mozilla.org/apps/fenix/metrics/browser_search_with_ads), then we need to update the [documentation in the registry file](https://github.com/mozilla-mobile/fenix/blob/4f5165d1da9d27076c6e849600d34bcd40dba340/app/metrics.yaml#L6202).

Reply:
Opened [#28010](https://github.com/mozilla-mobile/fenix/issues/28010) in this scope

Reply:
Ok, ideally you'd want that documentation change as part of this PR, but I'm fine with having that separately for this one time :)